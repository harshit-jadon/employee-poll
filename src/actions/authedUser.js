export const EMPLOYEE_USER = "EMPLOYEE_USER";
export const LOGOUT_EMPLOYEE = "LOGOUT_EMPLOYEE";

export function employeeUser(authedUser) {
  return {
    type: EMPLOYEE_USER,
    authedUser,
  };
}

export function logoutEmp() {
  return {
    type: LOGOUT_EMPLOYEE,
  };
}

export function loginToApp(username, password) {
  return (dispatch, getState) => {
    const { users } = getState();
    const user = Object.values(users).find(
      (user) => user.id === username && user.password === password
    );
    if (user) {
      return dispatch(employeeUser(user));
    }
  };
}

export function logoutApp() {
  return (dispatch) => {
    return dispatch(logoutEmp());
  };
}
