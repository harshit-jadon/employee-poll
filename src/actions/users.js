export const ADD_QUESTION_EMPL = "ADD_QUESTION_EMPL";
export const ADD_ANSWER_EMPL = "ADD_ANSWER_EMPL";
export const RECEIVE_EMPL = "RECEIVE_EMPL";

export function addQuestionEmpl({ author, id }) {
  return {
    type: ADD_QUESTION_EMPL,
    author,
    qid: id,
  };
}

export function addAnswerEmpl(authedUser, qid, answer) {
  return {
    type: ADD_ANSWER_EMPL,
    authedUser,
    qid,
    answer,
  };
}

export function receiveEmpl(users) {
  return {
    type: RECEIVE_EMPL,
    users,
  };
}