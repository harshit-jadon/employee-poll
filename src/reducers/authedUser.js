import { LOGOUT_EMPLOYEE, EMPLOYEE_USER } from "../actions/authedUser";

export default function authedUser(state = null, action) {
  switch (action.type) {
    case EMPLOYEE_USER:
      return action.authedUser;
    case LOGOUT_EMPLOYEE:
      return null;
    default:
      return state;
  }
}
