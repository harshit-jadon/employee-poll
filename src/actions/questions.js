import { saveQuestion, saveQuestionAnswer } from "../util/api";
import { addAnswerEmpl, addQuestionEmpl } from "./users";

export const ADD_QUESTION_ANSWER = "ADD_QUESTION_ANSWER";
export const ADD_ONLY_QUESTION = "ADD_ONLY_QUESTION";
export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";


function addQuestionAnswer(author, qid, answer) {
  return {
    type: ADD_QUESTION_ANSWER,
    author,
    qid,
    answer,
  };
}

function addOnlyQuestion(question) {
  return {
    type: ADD_ONLY_QUESTION,
    question,
  };
}

export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}


export function addAnswerFnc(questionId, answer) {
  return (dispatch, getState) => {
    const { authedUser } = getState();
    return saveQuestionAnswer(authedUser.id, questionId, answer).then(() => {
      dispatch(addQuestionAnswer(authedUser.id, questionId, answer));
      dispatch(addAnswerEmpl(authedUser.id, questionId, answer));
    });
  };
}

export function addQuestionFnc(firstOption, secondOption) {
  return (dispatch, getState) => {
    const { authedUser } = getState();
    return saveQuestion(firstOption, secondOption, authedUser).then(
      (question) => {
        dispatch(addOnlyQuestion(question));
        dispatch(addQuestionEmpl(question));
      }
    );
  };
}


