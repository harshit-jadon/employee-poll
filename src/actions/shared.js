import { receiveEmpl } from "./users";
import { receiveQuestions } from "./questions";
import { getInitialData } from "../util/api";

export function handleDataInitial() {
  return (dispatch) => {
    return getInitialData().then(({ users, questions }) => {
      dispatch(receiveEmpl(users));
      dispatch(receiveQuestions(questions));
    });
  };
}
