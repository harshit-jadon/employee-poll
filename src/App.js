import React, { useEffect } from "react";
import "./App.css";
import { Route, Routes } from "react-router-dom";
import { connect } from "react-redux";
import LoginPage from "../src/components/loginPage/LoginPage";
import PrivateRoute from "../src/components/privateRoute/PrivateRoute";
import EmployeeDashboard from "../src/components/employeeDashboard/EmployeeDashboard";
import NavBar from "../src/components/navBar/NavBar";
import NewPollPage from "../src/components/newPollPage/NewPollPage";
import Leaderboard from "./components/leaderBoard/Leaderboard";
import PollPage from "../src/components/pollPage/PollPage";
import Error404 from "./components/404";
import { handleDataInitial } from "./actions/shared";

function App({ dispatch, login }) {

  useEffect(() => {
    dispatch(handleDataInitial());
  });

  return (
    <div>
      {login && <NavBar />}
      <Routes>
        <Route path="/login" exact element={<LoginPage />} />
        <Route
          path="/"
          element={
            <PrivateRoute>
              <EmployeeDashboard />
            </PrivateRoute>
          }
        />
        <Route
          path="/add"
          exact
          element={
            <PrivateRoute>
              <NewPollPage />
            </PrivateRoute>
          }
        />
        <Route
          path="/leaderboard"
          exact
          element={
            <PrivateRoute>
              <Leaderboard />
            </PrivateRoute>
          }
        />
        <Route
          path="/questions/:id"
          element={
            <PrivateRoute>
              <PollPage />
            </PrivateRoute>
          }
        />

        <Route path="/404" exact element={<Error404 />} />
        <Route path="/*" exact element={<Error404 />} />
      </Routes>
    </div>
  );
}

const mapStateToProps = ({ authedUser }) => ({
  login: authedUser,
});

export default connect(mapStateToProps)(App);
