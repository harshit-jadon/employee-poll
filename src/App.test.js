import React from 'react';
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {render} from '@testing-library/react';
import {store} from "./store";
import userEvent from '@testing-library/user-event';
import {setAuthedUser} from "./actions/authedUser";
import App from './App';

describe("App.js component", () => {

    it("should render the component app.js", () => {
        const Appcomponent = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        expect(Appcomponent).toBeDefined();
        expect(Appcomponent).toMatchSnapshot();
    });

    it("should render the loginpage component", () => {
        const loginPageComponent = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        const loginpageComp = loginPageComponent.getByTestId("login-header");
        expect(loginpageComp).toBeInTheDocument();
    });

    it("should show Dashboard page when logged in", () => {
        store.dispatch(setAuthedUser({id: "", password: ""}));
        const dashboardComp = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        const employeeDashboard = dashboardComp.getByTestId("employee-dashboard");
        expect(employeeDashboard).toBeInTheDocument();
    });


});
