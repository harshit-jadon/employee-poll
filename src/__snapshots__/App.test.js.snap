// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js component should render the component app.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="head-container"
        >
          <h1
            class="header"
            data-testid="login-header"
          >
            Employee Polls
          </h1>
          <img
            alt="login-page-svg"
            src="employee1.png"
          />
          <h1>
            Log In
          </h1>
          <form
            class="form-div"
          >
            <div
              class="div-container"
            >
              <label
                class="label-box"
                for="user"
              >
                User
              </label>
              <div>
                <input
                  class="input-box"
                  data-testid="username-input"
                  placeholder="User"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="div-container"
            >
              <label
                class="label-box"
                for="password"
              >
                Password
              </label>
              <div
                class="mt-1"
              >
                <input
                  class="input-box"
                  data-testid="password-input"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="submit-btn"
            >
              <button
                class="submit"
                data-testid="submit-button"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="head-container"
      >
        <h1
          class="header"
          data-testid="login-header"
        >
          Employee Polls
        </h1>
        <img
          alt="login-page-svg"
          src="employee1.png"
        />
        <h1>
          Log In
        </h1>
        <form
          class="form-div"
        >
          <div
            class="div-container"
          >
            <label
              class="label-box"
              for="user"
            >
              User
            </label>
            <div>
              <input
                class="input-box"
                data-testid="username-input"
                placeholder="User"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="div-container"
          >
            <label
              class="label-box"
              for="password"
            >
              Password
            </label>
            <div
              class="mt-1"
            >
              <input
                class="input-box"
                data-testid="password-input"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="submit-btn"
          >
            <button
              class="submit"
              data-testid="submit-button"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
